src/app/services/http.service.ts

```diff
@@ -1,45 +1,68 @@
 import { HttpClient, HttpErrorResponse, HttpHeaders } from '@angular/common/http';
 import { Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material/snack-bar';
 import { Observable, throwError } from 'rxjs';
 import { environment } from 'src/environments/environment';
-import { User } from '../models/user';
 
 @Injectable({
   providedIn: 'root'
 })
 export class HttpService {
+  headers!: HttpHeaders;
 
   constructor(private httpClient: HttpClient) {
 
   }
   
-  makeUserHttpRequest(requestName: string, requestBody?: any, ...pathParameters: string[]): Observable<any> {
-    const requestUrl: string = this.buildUrlPath(...pathParameters);
+  getAll<T>(endpoint: string): Observable<T[]> {
+    this.setRequestHeaders();
+    const requestUrl = this.getRequestUrl(endpoint);
 
-    const headers = new HttpHeaders( {
-      'Content-Type': 'applicaton/json'
-    })
+    return this.httpClient.get<T[]>(requestUrl, { headers: this.headers });
+  }
 
-    switch (requestName) {
-      case 'post': return this.userPostRequest(requestUrl, requestBody, headers);
-      default:  return throwError( () => new Error('Invalid Http Request.'));
-    }
+  getById<T>(id: number, endpoint: string): Observable<T> {
+    this.setRequestHeaders();
+    const requestUrl = this.getRequestUrl(endpoint) + '/' + id;
 
+    return this.httpClient.get<T>(requestUrl, { headers: this.headers });
   }
 
-  private userPostRequest(requestUrl: string, requestBody?: any, headers?: any): Observable<any> {
-    return this.httpClient.post<User>(requestUrl, requestBody, headers);
+  post<T>(input: any, endpoint: string): Observable<T> {
```

Comment:
what does this "input" parameter represent? could we have a suitable name for it?

Reply:
Here, input represent the body of the request.
Should I use something along the line of 'requestBody' instead?